libslide/src/partial_evaluator.rs

```diff
@@ -1,39 +1,33 @@
 use crate::evaluator_rules::*;
 use crate::grammar::*;
+use crate::utils::hash;
 
-use core::hash::{Hash, Hasher};
-use std::collections::{hash_map::DefaultHasher, HashSet};
+use std::collections::HashSet;
+use std::rc::Rc;
 
 /// Evaluates an expression to as simplified a form as possible.
 /// The evaluation may be partial, as some values (like variables) may be unknown.
 pub fn evaluate(expr: Stmt) -> Expr {
     let rule_set = RuleSet::default();
     let built_rules = rule_set.build();
 
-    let mut simplified_expr: Expr = match expr {
-        Stmt::Expr(expr) => expr,
+    let mut simplified_expr: Rc<Expr> = match expr {
+        Stmt::Expr(expr) => expr.into(),
         _ => todo!("Evaluation currently only handles expressions"),
     };
 
```

Comment:
add a // todo comment here so we are consistent with our other todos

Reply:
will do